General updating

genebean opened the pull request:
This is built atop #9 and includes its change. This provides full test coverage for the manifests but still does not cover the ruby code.

genebean committed:
Testing and formatting updates

awrenn commented on manifests/app_stack.pp:
```diff
@@ -331,50 +325,36 @@
-        owner  => $_final_hdp_user,
-        group  => $_final_hdp_user,
-        ;
+    $_minio_directories = [
```
Much better

awrenn commented on spec/classes/app_stack_spec.rb:
```diff
@@ -42,17 +77,23 @@
+                'File[/tmp/ui-ca.pem]',
+                'File[/tmp/ui-cert.key]',
+                'File[/tmp/ui-cert.pem]',
+                'File[/opt/puppetlabs/hdp/docker-compose.yaml]',
```
Asking for mostly learning puppet purposes - this docker_compose resource requires only the docker-compose file, but subscribes to the cert and key files as well. Should it also require these files if its subscribed to them? Or because it doesn't require them under certain config options, we don't include that here?

genebean replied:
That is a very good and valid question. That was being tested with these lines from above:

```ruby
        it { is_expected.to  contain_file('/tmp/ui-cert.key').that_comes_before('Docker_compose[hdp]') }
        it { is_expected.to  contain_file('/tmp/ui-cert.pem').that_comes_before('Docker_compose[hdp]') }
```

I have made this more clear by combining those lines into this block and will push it in just a moment

genebean replied:
This change has been pushed here

genebean committed:
UI TLS doesn't need CA set

It can be set, but is not required when just adding certs for the
frontend web interface
